src/class-elements/private-methods/cls-decl-export-default.template

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2019 Jaideep Bhoosreddy (Bloomberg LP). All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+path: language/statements/class/elements/private-methods/export-default-
+name: private method definitions in a class declaration
+flags: [module]
+features: [class, class-methods-private]
+info: |
+  ClassElement :
+    MethodDefinition
+    ...
+    ;
+
+  ClassElementName :
+    PropertyName
+    PrivateName
+
+  PrivateName ::
+    # IdentifierName
+
+  MethodDefinition :
+    ClassElementName ( UniqueFormalParameters ) { FunctionBody }
+    GeneratorMethod
+    AsyncMethod
+    AsyncGeneratorMethod
+    get ClassElementName () { FunctionBody }
+    set ClassElementName ( PropertySetParameterList ) { FunctionBody }
+
+  GeneratorMethod :
+    * ClassElementName ( UniqueFormalParameters ){GeneratorBody}
+
+  AsyncMethod :
+    async [no LineTerminator here] ClassElementName ( UniqueFormalParameters ) { AsyncFunctionBody }
+
+  AsyncGeneratorMethod :
+    async [no LineTerminator here]* ClassElementName ( UniqueFormalParameters ) { AsyncGeneratorBody }
+
+  ---
+
+  InitializeClassElements ( F, proto )
+
+  ...
+  5. For each item element in order from elements,
+    a. Assert: If element.[[Placement]] is "prototype" or "static", then element.[[Key]] is not a Private Name.
+    b. If element.[[Kind]] is "method" and element.[[Placement]] is "static" or "prototype",
+      i. Let receiver be F if element.[[Placement]] is "static", else let receiver be proto.
+      ii. Perform ? DefineClassElement(receiver, element).
+
+  InitializeInstanceElements ( O, constructor )
+
+  ...
+  3. Let elements be the value of F's [[Elements]] internal slot.
+  4. For each item element in order from elements,
+    a. If element.[[Placement]] is "own" and element.[[Kind]] is "method",
+      i. Perform ? DefineClassElement(O, element).
+
+  DefineClassElement (receiver, element)
+
+  ...
+  6. If key is a Private Name,
+    a. Perform ? PrivateFieldDefine(receiver, key, descriptor).
+
+  PrivateFieldDefine (P, O, desc)
+
+  ...
+  6. Append { [[PrivateName]]: P, [[PrivateFieldDescriptor]]: desc } to O.[[PrivateFieldDescriptors]].
+
+esid: prod-MethodDefinition
+---*/
+
+/*** template notes
+ * method should always be #m
+ * the template provides c.ref() for external reference
+ */
+
+function hasProp(obj, name, expected, msg) {
```

Comment:
I'm curious to know why this is here and you're not using `verifyProperty` via `includes: [propertyHelper.js]`

Reply:
this is from the other templates we already have and could be improved. hasProp is always used for non existing own properties, and verifyProperty is meant for expected to exist properties.

this could be shortened to `hasNoProp(obj, name, msg)` and doing the following assertions:

```js
function hasNoProp(obj, name, msg) {
  var hasOwnProperty = Object.prototype.hasOwnProperty.call(obj, name);
  assert.sameValue(hasOwnProperty, false, msg);

  var hasProperty = Reflect.has(obj, name);
  assert.sameValue(hasProperty, false, msg);

// or `name in obj` instead of Reflect.has
}
```

Reply:
in any case, this should observe any conclusion for https://github.com/tc39/test262/pull/2197#issuecomment-503722953

Reply:
Can we move `hasProp` to `propertyHelper.js` so there isn't many copies of it floating around?

Reply:
yes, but this needs to be done in another PR for the existing tests. It should be quick. I'm addressing it tomorrow.